src/main/java/com/marklogic/appdeployer/command/hosts/AssignHostsToGroupsCommand.java

```diff
@@ -0,0 +1,93 @@
+package com.marklogic.appdeployer.command.hosts;
+
+import java.util.Map;
+
+import com.marklogic.appdeployer.command.AbstractUndoableCommand;
+import com.marklogic.appdeployer.command.CommandContext;
+import com.marklogic.appdeployer.command.SortOrderConstants;
+import com.marklogic.mgmt.admin.ActionRequiringRestart;
+import com.marklogic.mgmt.resource.appservers.ServerManager;
+import com.marklogic.mgmt.resource.hosts.HostManager;
+
+public class AssignHostsToGroupsCommand extends AbstractUndoableCommand {
+
+	private static final String DEFAULT_GROUP_NAME = "Default";
+	private static final String MANAGE_SERVER_NAME = "Manage";
+	private static final String MANAGE_SERVER_JSON_TEMPLATE = "{\"server-name\":\"%s\", \"group-name\":\"%s\", \"server-type\":\"http\", \"root\":\"Apps/\", \"port\":8002, \"content-database\":\"App-Services\", \"error-handler\":\"manage/error-handler.xqy\", \"url-rewriter\":\"manage/rewriter.xqy\", \"privilege\":\"http://marklogic.com/xdmp/privileges/manage\"}";
+	private static final String APP_SERVER_NAME = "App-Services";
+	private static final String APP_SERVER_JSON_TEMPLATE = "{\"server-name\":\"%s\", \"group-name\":\"%s\", \"server-type\":\"http\", \"root\":\"/\", \"port\":8000, \"modules-database\":\"Modules\", \"content-database\":\"Documents\", \"error-handler\":\"/MarkLogic/rest-api/8000-error-handler.xqy\", \"url-rewriter\":\"/MarkLogic/rest-api/8000-rewriter.xml\", \"rewrite-resolves-globally\":true}";
+
+	public AssignHostsToGroupsCommand() {
+        setExecuteSortOrder(SortOrderConstants.ASSIGN_HOSTS_TO_GROUPS);
+        setUndoSortOrder(SortOrderConstants.UNASSIGN_HOSTS_FROM_GROUPS);
+    }
+
+	@Override
+	public void execute(CommandContext context) {
+		context.getAdminManager().invokeActionRequiringRestart(new ActionRequiringRestart() {
```

Comment:
Fun Java 8 trick - you can use a lambda here, will look like this:

    context.getAdminManager().invokeActionRequiringRestart(() -> assignHostsToGroups(context));